docs/advanced/document-store/elasticsearch.md

```diff
@@ -118,6 +118,103 @@ da2.summary()
 
 Other functions behave the same as in-memory DocumentArray.
 
+
+### Search with `text`  
+
+Text search can be easily leveraged in a `DocumentArray` with `storage='elasticsearch'`.
+To do this text needs to be indexed using the boolean flag `'index_text'` which is set when
+the `DocumentArray` is created  with `config={'index_text': True, ...}`.  
+The following example builds a `DocumentArray` with several documents containing text and searches
+for those that have `pizza` in their text description.
+
+```python
+from docarray import DocumentArray, Document
+
+da = DocumentArray(storage='elasticsearch', config={'n_dim': 2, 'index_text': True})
+da.extend(
+    [
+        Document(id='1', text='Person eating'),
+        Document(id='2', text='Person eating pizza'),
+        Document(id='3', text='Pizza restaurant'),
+    ]
+)
+
+pizza_docs = da.find('pizza')
+pizza_docs[:, 'text']
+```
+will print
+```text
+['Pizza restaurant', 'Person eating pizza']
+```
+
+### Search with `tags`
+
+Text can also be indexed when it is part of `tags`.
+This is mostly useful in applications where text data can be split into groups and applications might require 
+retrieving items based on a text search in an specific tag.
+
+For example:
+```python
+from docarray import DocumentArray, Document
+
+da = DocumentArray(
+    storage='elasticsearch', config={'n_dim': 32, 'tag_indices': ['food_type', 'price']}
+)
+da.extend(
+    [
+        Document(
+            id='1',
```

Comment:
if this `id` is not required for reproducibility, i suggest we remove it. For the sake of clarity: we do not recommend manually coded `id`